knowledge-base/adr/0015-date-time-handling.md

```diff
@@ -61,3 +61,92 @@ Declined, as we are planning to release a version 2 of the SDK in the first half
 
 ### Implement SDK Date wrapper that abstracts different libraries
 Declined, because this would probably mean quite high implementation effort and is not flexible enough.
+
+## Summary of the [PoC](https://github.com/SAP/cloud-sdk-js/pull/921)
+### New Api design
+Instead of the original idea of creating a new function `transform`, the middleware will be passed via the existing functions as optional parameter when building an `entity` or a `requestBuilder`.
+See examples below:
+```ts
+// Default entity builder usage
+const a = TestEntity.builder().int32Property(1).build();
+// Default create request
+TestEntity.requestBuilder().create(a).execute();
+
+// Transformed entity builder usage
+const b = TestEntity.builder(serializers).int32Property('test').build();
+// Transformed create request
+TestEntity.requestBuilder(serializers).create(b).execute();
+```
+The middleware has a structure like below to define (de)serializers for some edm types.
+```ts
+interface DeSerializationMiddlewareInterface<T1>{
+  'Edm.String'?: {
+    serializer: (val: T1) => string;
+    deserializer: (ori: string) => T1;
+  };
+  'Edm.Int32'?: {
+     serializer: (val: T1) => string;
+     deserializer: (ori: string) => T1;
+  };
+
+}
+```
+**Please note**
+
+With the api design above, when applying it for replacing the moment lib, we will introduce multiple generic types for `Entity`, `RequestBuilder`, `TestEntity` and `TestEntityRequestBuilder`, which makes our code more complicated, but the users should not be affected.
+The number of generic types depends on the number of related edm types. 
+For example, for `ODataV4`, 4 edm types (`Edm.DateTimeOffset`, `Edm.Date`, `Edm.Duration` and `Edm.TimeOfDay`) are relevant, which means it will introduce 4 generic types for all the classes mentioned above.
+
+### Assumption
+- Request builders has the same complexity.
+- Serializer has the same complexity as the Deserializer.
+- OData V2 has the same complexity as the OData V4.
+
+### Scope
```

Comment:
One important part of the scope is missing IMO: We want to save space so if you use a non moment middle ware the moment dependency should not appear at all in your dependencies. Add a paragraph how this is achieved.